aptos-move/framework/aptos-framework/sources/transaction_validation.move

```diff
@@ -728,8 +903,10 @@ module aptos_framework::transaction_validation {
             };
         };
 
-        // Increment sequence number
-        let addr = signer::address_of(&account);
-        account::increment_sequence_number(addr);
+        if (!is_orderless_txn) {
+            // Increment sequence number
+            let addr = signer::address_of(&account);
+            account::increment_sequence_number(addr);
+        }
```

Comment:
@georgemitenkov , @vgao1996 - does the following logic work from gas perspective:
- for seq_num txns, we check in prologue, but increment in epilogue
- for orderless we check and modify in prologue.

if second one is OK for orderless, should we do the same for seq_num, and move this to prologue?

Reply:
this would require a feature flag though since it changes the gas behavior (seq num mutation now will be charged)

Reply:
Changes in move code dont require feature flag,  as they are gated by framework upgrade.

But- when txn aborts- is prologue also reverted? If that's the case then these wild need to go to epilogue

Reply:
prologue is reverted too. but gas charging is not in move code, which this will change (epilogue is not charged but prologue is)

Reply:
The epilogue session for aborted transactions is initiated with the prologue_change_set/abort_hook_change_set as input [here](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/aptos-vm/src/aptos_vm.rs#L667).
When the epilogue returns its change set, it first squashes with previous change set [here](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/aptos-vm/src/move_vm_ext/session/respawned_session.rs#L96-L104).

When a transaction is aborted, the prologue's change set is included in the final change set of the transaction right? So, prologue is not reverted?

Reply:
@vusirikala is right, the prologue is not reverted - on txn abort its change set is used to shadow the original state and the writes are later included in txn outputs.

@igor-aptos I think incrementing sequence number in prologue makes sense. One note: if this is a sponsored account creation txn, we create the account AFTER we run the prologue. So we need to change that to be able to  have a sequence number to increment?